test/opt/local_ssa_elim_test.cpp

```diff
@@ -2024,6 +2024,788 @@ OpFunctionEnd
   SinglePassRunToBinary<SSARewritePass>(text, false);
 }
 
+TEST_F(LocalSSAElimTest, DebugForLoop) {
+  // #version 140
+  //
+  // in vec4 BC;
+  // out float fo;
+  //
+  // void main()
+  // {
+  //     float f = 0.0;
+  //     for (int i=0; i<4; i++) {
+  //       f = f + BC[i];
+  //     }
+  //     fo = f;
+  // }
+
+  const std::string text = R"(
+; CHECK: [[f_name:%\w+]] = OpString "f"
+; CHECK: [[i_name:%\w+]] = OpString "i"
+; CHECK: [[dbg_f:%\w+]] = OpExtInst %void [[ext:%\d+]] DebugLocalVariable [[f_name]]
+; CHECK: [[dbg_i:%\w+]] = OpExtInst %void [[ext]] DebugLocalVariable [[i_name]]
+
+; CHECK:      OpStore %f %float_0
+; CHECK:      OpStore %i %int_0
+; CHECK:      OpExtInst %void [[ext]] DebugDeclare [[dbg_f]] %f
+; CHECK:      OpExtInst %void [[ext]] DebugDeclare [[dbg_i]] %i
```

Comment:
Is there a DebugValue after the initialization of `f` and `i`?  If not, there should be.  Otherwise, we will have to keep the variable around otherwise the debugger won't know what the value is until it reaches the first debug value.

I'm wondering if we should be keeping the debug delcare around if the ssa rewriter removes the use of the variable.  Do you know what happens in LLVM?

Reply:
LLVM gets rid of the alloca as well, which is a different design decision that what we did.  Diego decided to keep the stores, so the OpVariable instruction must stick around.

I'm leaning towards removing the debug declare in ssa-rewrite, but we should have a quick discussion about it tomorrow with Diego.